osbrain/tests/test_proxy.py

```diff
@@ -119,6 +122,48 @@ def ping(self):
         InitTimeoutProxy('foo', timeout=1.)
 
 
+@pytest.mark.parametrize('timeout', [-.5, 0., .5])
+def test_agent_proxy_wait_running(nsproxy, timeout):
+    """
+    Using `wait_for_running` on a proxy after initialization should block until
+    the agent is running or time out.
+    """
+    ap = AgentProcess('agent')
+    ap.start()
+
+    # Get "offline" proxy
+    agent = Proxy('agent')
+    time0 = time.time()
+    Timer(abs(timeout)/2, agent.run).start()
+
+    # Wait long enough for the agent to start running
+    proxy = Proxy('agent').wait_for_running(timeout=timeout)
+    assert proxy.ping() == 'pong'
+    if timeout != 0:
+        assert abs(timeout)/2 <= time.time() - time0 <= abs(timeout)
```

Comment:
I liked the `0.` timeout case in a separate case more, just like you had it before. I think the negative case should be in a separate case as well. Just maybe.